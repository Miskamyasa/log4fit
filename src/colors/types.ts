import {ColorSchemeName as SchemeName} from "react-native"

export type ColorNames =
  | "text"
  | "viewBackground"
  | "dimmedBackground"
  | "buttonBackground"
  | "buttonText"
  | "screenBackground"
  | "headerTitle"
  | "headerBackground"
  | "loaderColor"
  | "loaderBackground"
  | "overlayBackground"
  | "dividerColor"
  | "alwaysWhite"

export type ColorSchemeName = NonNullable<SchemeName>

export type ThemeProps = {
    light: string
    dark: string
}
