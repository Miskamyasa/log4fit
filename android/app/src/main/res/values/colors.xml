<resources>
  <color name="iconBackground">#FFFFFF</color>
  <color name="colorPrimary">#023c69</color>
  <color name="colorPrimaryDark">#000000</color>
</resources>