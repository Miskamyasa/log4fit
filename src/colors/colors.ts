import {ColorNames, ColorSchemeName} from "./types";


type Colors = Record<ColorSchemeName, Record<ColorNames, string>>;

export const mainBackgroundColor = "#000";

const header = {
  headerBackground: mainBackgroundColor,
  headerTitle: "#fefefe",
};

export const schemes: Colors = Object.freeze({
  light: {
    text: "#333",
    viewBackground: "#fefefe",
    buttonBackground: "#f7f8f9",
    buttonText: "#678",
    screenBackground: "#fefefe",
    loaderBackground: "transparent",
    loaderColor: "#248",
    gotoWorkoutText: "#245",
    gotoWorkoutBackground: "rgba(210, 220, 230, 0.82)",
    overlayBackground: "rgba(210, 220, 230, 0.93)",
    ...header,
  },
  dark: {
    text: "#fefefe",
    viewBackground: mainBackgroundColor,
    buttonBackground: "#070809",
    buttonText: "#f7f8f9",
    screenBackground: mainBackgroundColor,
    loaderBackground: "transparent",
    loaderColor: "#f2f4f8",
    gotoWorkoutText: "#f1f4f9",
    gotoWorkoutBackground: "rgba(0, 0, 0, 0.7)",
    overlayBackground: "rgba(0, 0, 0, 0.89)",
    ...header,
  },
});
