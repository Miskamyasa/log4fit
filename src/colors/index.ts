export * from "./types";
export * from "./colors";
export * from "./useColorScheme";
export * from "./useThemeColor";
