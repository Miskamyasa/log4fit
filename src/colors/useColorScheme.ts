import {useColorScheme as _useColorScheme} from "react-native"

import type {ColorSchemeName} from "./types"

export function useColorScheme(): ColorSchemeName {
    return _useColorScheme() ?? "dark"
}
