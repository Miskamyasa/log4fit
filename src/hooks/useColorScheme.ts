import {ColorSchemeName as SchemeName, useColorScheme as _useColorScheme} from "react-native";

export type ColorSchemeName = NonNullable<SchemeName>;

export function useColorScheme(): ColorSchemeName {
  return _useColorScheme() || "dark";
}
