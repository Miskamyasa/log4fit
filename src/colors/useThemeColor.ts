import {schemes} from "./colors"
import {ThemeProps, ColorNames} from "./types"
import {useColorScheme} from "./useColorScheme"


export function useThemeColor(colorName: ColorNames, props?: ThemeProps): string {
  const colorScheme = useColorScheme()
  return props && props[colorScheme] || schemes[colorScheme][colorName]
}
