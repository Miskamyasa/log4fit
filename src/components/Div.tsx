import {ElementType, memo, ReactNode, useMemo} from "react"
import {View, TouchableOpacity, ViewStyle} from "react-native"

import {ColorNames, ThemeProps} from "../colors/types"
import {useThemeColor} from "../colors/useThemeColor"

export const Div = memo(function Div(props: {
    children: ReactNode
    style?: ViewStyle | ViewStyle[]
    colorName?: ColorNames
    onPress?: () => void
    theme?: ThemeProps
    disabled?: boolean
}) {
    const {style, theme, colorName = "viewBackground", onPress, disabled, children} = props
    const backgroundColor = useThemeColor(colorName, theme)

    const Container = onPress ? TouchableOpacity : View as ElementType

    const styles = useMemo(() => {
        const styles: ViewStyle[] = [{backgroundColor, opacity: disabled ? 0.61 : 1}]
        if (style) {
            Array.isArray(style)
                ? styles.push(...style)
                : styles.push(style)
        }
        return styles
    }, [backgroundColor, disabled, style])

    return (
        <Container
            disabled={disabled}
            style={styles}
            onPress={onPress}>
            {children}
        </Container>
    )
})
