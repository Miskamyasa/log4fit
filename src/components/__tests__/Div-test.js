import {Text} from "react-native";
import renderer from "react-test-renderer";

import {useColorScheme} from "../../colors/useColorScheme";
import Div from "../Div";


jest.mock("../../colors/useColorScheme");

describe("components: View", () => {
  const DEFAULT = "light";
  useColorScheme.mockReturnValue(DEFAULT);

  const children = <Text>View</Text>;

  // const {result: colorScheme} = renderHook(() => useColorScheme());

  it("renders with default (light theme)", () => {
    const tree = renderer.create(<Div>{children}</Div>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with dark theme", () => {
    const DARK = "dark";
    useColorScheme.mockReturnValue(DARK);
    const tree = renderer.create(<Div>{children}</Div>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with custom background color #AAA", () => {
    const background = "#AAA";
    const props = {light: background, dark: background};
    const tree = renderer.create(<Div {...props}>{children}</Div>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with custom styles {padding: 12}", () => {
    const props = {style: {padding: 12}};
    const tree = renderer.create(<Div {...props}>{children}</Div>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
