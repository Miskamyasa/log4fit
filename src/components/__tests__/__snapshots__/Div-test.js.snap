// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components: View renders with custom background color #AAA 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#000",
        "opacity": 1,
      },
    ]
  }
>
  <Text>
    View
  </Text>
</View>
`;

exports[`components: View renders with custom styles {padding: 12} 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#000",
        "opacity": 1,
      },
      {
        "padding": 12,
      },
    ]
  }
>
  <Text>
    View
  </Text>
</View>
`;

exports[`components: View renders with dark theme 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#000",
        "opacity": 1,
      },
    ]
  }
>
  <Text>
    View
  </Text>
</View>
`;

exports[`components: View renders with default (light theme) 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fdfdfe",
        "opacity": 1,
      },
    ]
  }
>
  <Text>
    View
  </Text>
</View>
`;
